amulet/operations/fill.py

```diff
@@ -19,4 +19,5 @@ def fill(world: "World", target_box: Selection, options: dict):
     internal_id = world.palette.get_add_block(fill_block)
 
     for chunk, slices, _ in world.get_chunk_slices(target_box):
-        chunk.blocks[slices] = internal_id
+        chunk.blocks2[slices] = internal_id
+        chunk.changed = True
```

Comment:
are you talking about `chunk.changed = True`?
The old implementation used to automatically do that in the chunk array but there are multiple ways to access the data now and it just seems cleaner to do it manually in the operation especially since there are some places where we can't automatically mark the chunk as changed.

Reply:
Ideally from efficiency point of view it would be nice to have "is changed" flag per section but that's just nice to have for faster saving and it wouod be a bit I convenient for normal use.

Reply:
Our workflow loads the whole chunk from disk into the Chunk class.
When saving we check that flag to see if the chunk should be saved back (if it is false we leave it as it was on disk). We then recreate all the nbt data from the Chunk class so for the vanilla save format it makes no difference (and is probably slower) to have a changed flag per sub-chunk.

It might help in the render since you can rebuild a single sub-chunk rather than the whole chunk but that would require even more of a rewrite in the renderer.